### PR TITLE
chore(IDX): split .bazelrc

### PR DESCRIPTION
**Pull request by nmattia**
This splits the `.bazelrc` into different fragment files: one with all necessary build options, and one with DFINITY-internal configuration (like cache, metrics upload, etc). This will help build the repo without access to DFINITY infrastructure.

### CODE REVIEW

File: bazel/conf/.bazelrc.build [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+test --test_output=errors
+test --test_env=RUST_BACKTRACE=full
+
+test:precommit --build_tests_only --test_tag_filters="smoke"
+
+build:systest --build_tag_filters= --s3_endpoint=https://s3-upload.idx.dfinity.network
+run:systest --remote_upload_local_results=true
+test:systest --remote_upload_local_results=true --test_output=streamed --test_tag_filters=
+
+build:testnet --build_tag_filters= --s3_endpoint=https://s3-upload.idx.dfinity.network --ic_version_rc_only=
```
Comment (marko-k0) [inline bazel/conf/.bazelrc.build:66]: This should go under internal as well.

File: bazel/conf/.bazelrc.build [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+test:alltests --test_tag_filters=""
+test:paritytests --test_tag_filters="-system_test"
+test:ci --test_tag_filters="-post_master,-system_test_hourly,-system_test_nightly,-system_test_nightly_nns,-system_test_staging,-system_test_hotfix,-fuzz_test,-nns_tests_nightly"
+
+test --test_output=errors
+test --test_env=RUST_BACKTRACE=full
+
+test:precommit --build_tests_only --test_tag_filters="smoke"
+
+build:systest --build_tag_filters= --s3_endpoint=https://s3-upload.idx.dfinity.network
```
Comment (marko-k0) [inline bazel/conf/.bazelrc.build:62]: This should go under internal as well.

Comment (marko-k0) [inline bazel/conf/.bazelrc.build:62]: This should go under internal as well.

File: bazel/conf/.bazelrc.build (hunk 1 above)
Comment (nmattia) [reply bazel/conf/.bazelrc.build:66]: good catch!

File: bazel/conf/.bazelrc.internal [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+build:local --experimental_remote_downloader=
+build --remote_local_fallback
+build    --remote_upload_local_results=false
+build:ci --remote_upload_local_results=true
+
+build:ci --noremote_local_fallback
+build:ci --build_tag_filters="-system_test,-fuzz_test"
+build:ci --verbose_failures
+
+# Run `bazel build ... --config=local` if you don't have access to the buildfarm infrastructure.
```
Comment (marko-k0) [inline bazel/conf/.bazelrc.internal:36]: This section probably fits more to `.bazelrc.build`, no?

Comment (nmattia) [reply bazel/conf/.bazelrc.internal:36]: not really; the `--config=local` only applies to the "internal" config (i.e. if you have the remote cache set up but don't want to use it)

**Review by marko-k0 — APPROVED** [review]

File: bazel/conf/.bazelrc.internal (hunk 3 above)
Comment (nmattia) [reply bazel/conf/.bazelrc.internal]: updated the comment to make more sense now that the config is plit

File: bazel/conf/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Bazel configuration
+
+This directory contains some bazelrc fragments that are used in different scenarios. The fragments
+can be used like this:
+
+```
+bazel build --bazelrc <fragment> --bazelrc <fragment>
```
Comment (nmattia) [inline bazel/conf/README.md:7]:
```suggestion
bazel build --noworkspace_rc --bazelrc <fragment> --bazelrc <fragment>
```

**Pull request merged**